compiler/src/dotty/tools/io/Path.scala

```diff
@@ -173,64 +176,68 @@ class Path private[io] (val jfile: JFile) {
   def addExtension(ext: String): Path = Path(path + "." + ext)
   // changes the existing extension out for a new one, or adds it
   // if the current path has none.
-  def changeExtension(ext: String): Path = (
+  def changeExtension(ext: String): Path =
     if (extension == "") addExtension(ext)
     else Path(path.stripSuffix(extension) + ext)
-  )
 
   // conditionally execute
   def ifFile[T](f: File => T): Option[T] = if (isFile) Some(f(toFile)) else None
   def ifDirectory[T](f: Directory => T): Option[T] = if (isDirectory) Some(f(toDirectory)) else None
 
   // Boolean tests
-  def canRead = jfile.canRead()
-  def canWrite = jfile.canWrite()
-  def exists = try jfile.exists() catch { case ex: SecurityException => false }
-  def isFile = try jfile.isFile() catch { case ex: SecurityException => false }
+  def canRead = Files.isReadable(jpath)
+  def canWrite = Files.isWritable(jpath)
+  def exists = try Files.exists(jpath)  catch { case ex: SecurityException => false }
+  def isFile = try Files.isRegularFile(jpath)  catch { case ex: SecurityException => false }
   def isDirectory =
-    try jfile.isDirectory()
-    catch { case ex: SecurityException => jfile.getPath == "." }
-  def isAbsolute = jfile.isAbsolute()
+    try Files.isDirectory(jpath)
+    catch { case ex: SecurityException => jpath.toString == "." }
+  def isAbsolute = jpath.isAbsolute()
   def isEmpty = path.length == 0
 
   // Information
-  def lastModified = jfile.lastModified()
-  def length = jfile.length()
+  def lastModified = Files.getLastModifiedTime(jpath)
+  def length = Files.size(jpath)
 
   // Boolean path comparisons
   def endsWith(other: Path) = segments endsWith other.segments
   def isSame(other: Path) = toCanonical == other.toCanonical
-  def isFresher(other: Path) = lastModified > other.lastModified
+  def isFresher(other: Path) = lastModified.compareTo(other.lastModified) > 0
 
   // creations
   def createDirectory(force: Boolean = true, failIfExists: Boolean = false): Directory = {
-    val res = if (force) jfile.mkdirs() else jfile.mkdir()
+    val res = tryCreate(if (force) Files.createDirectories(jpath) else Files.createDirectory(jpath))
     if (!res && failIfExists && exists) fail("Directory '%s' already exists." format name)
     else if (isDirectory) toDirectory
-    else new Directory(jfile)
+    else new Directory(jpath)
   }
   def createFile(failIfExists: Boolean = false): File = {
-    val res = jfile.createNewFile()
+    val res = tryCreate(Files.createFile(jpath))
     if (!res && failIfExists && exists) fail("File '%s' already exists." format name)
     else if (isFile) toFile
-    else new File(jfile)
+    else new File(jpath)
   }
 
+  private def tryCreate(create: => JPath): Boolean =
+    try { create; true } catch { case _: FileAlreadyExistsException => false }
```

Comment:
Semantic is different for something like `foo/bar/` if `foo` does not exist

Reply:
As we saw together the semantics are the same. In the example, an `IOException` would be thrown in both implementations.